src/charge_controller/dev/ControllerUI.cpp

```diff
@@ -0,0 +1,106 @@
+#include <charge_controller/dev/ControllerUI.hpp>
+
+ControllerUI::ControllerUI(DEV::Encoder& encoder, DEV::Button& encoderButton, LCDDisplay& display, ControllerModel& model) : model(model), encoder(encoder), encoderButton(encoderButton), display(display) {
+    encoder.setRangeAndPosition(model.getEncoderRange(ControllerModel::State::PAGE_SELECT), 0);
+}
+
+void ControllerUI::process() {
+    switch (model.getState()) {
+    case ControllerModel::State::PAGE_SELECT:
+        switch (encoder.getPosition()) {
```

Comment:
```suggestion
        switch ((ControllerModel::Page) encoder.getPosition()) {
```
Although this doesn't change behavior, it makes things a bit more explicit

Reply:
I did not realize you could do that.